### DISCUSSION TITLE
How to get a database connection

### DISCUSSION THREAD

**Discussion by open-schnick** [rwf2/Rocket#2266] [Questions]
I wanted to know wheter or not it is possible to get a connection from the pool without the guard of an endpoint.
For example i want to access the database in a function that is not called from an endpoint.
Is there a way to get a connection from the pool?

**Comment by ELD** [comment]
The answer varies, a little, depending on whether you're using synchronous or asynchronous databases.

For `rocket_sync_db_pools`, there's a `get_one` method that's implemented on the database type. The one caveat is that it requires a `Rocket` instance to fetch the connection. That can be a little tricky getting an instance if the method is being called outside of a handler or other Rocket-related code.
Docs: https://api.rocket.rs/v0.5-rc/rocket_sync_db_pools/attr.database.html (not the `async fn get_one(...) ...` doc item)

For `rocket_db_pools`, there's a `fetch` method that returns the database instance. From there, you can access the inner type, grab a connection, and use it as you please. This also has the same caveat that you need a `Rocket` instance to fetch the database.
Docs: https://api.rocket.rs/v0.5-rc/rocket_db_pools/trait.Database.html#method.fetch

**Reply by open-schnick** [reply]
Ok thank you.
Is there a way to get a rocket instance available for all threads everywhere?
Do you have a code snippet or an example?
